llvm/lib/Target/AIE/AIE2TargetTransformInfo.cpp

```diff
@@ -27,6 +28,10 @@ static cl::opt<bool>
                 cl::desc("Enable true zero overhead hardware loops on AIE"),
                 cl::init(false), cl::Hidden);
 
+static cl::opt<int> MinIterCountHLReject(
+    "aie-hardware-loops-minitercount", cl::Hidden, cl::init(1),
+    cl::desc("Minimum trip count threshold for HL rejection"));
```

Comment:
Future: perhaps we can tie this with the "prepare-for-software-pipelining" pragma

Reply:
Hi @martien-de-jong , I think this is a good idea.